### PR TITLE
Add encoder for SPLADE

### PR DESCRIPTION
**Pull request by cadurosar**
Hi, here's a PR to add an encoder for SPLADE (not sure if it would work directly for the other sparse implems, but shouldn't be hard) and instructions for indexing and retrieving with Anserini. Most of it is based on the fact that one has already downloaded data for the coCondenser-marco example and has pre-installed Anserini.

### CODE REVIEW

**Comment by MXueguang** [comment]
Hi @cadurosar  Thanks for the PR. Do you have a record of the MRR@10 result by following the instruction?

**Comment by cadurosar** [comment]
Using monogpu (so per the instructions) I get: MRR@10: 35.90 R@1k 97.47%. 

I've also tried with 2 gpus and 4gpus (so using negatives_x_device, same batch size per gpu and torch.distributed.launch):

2_gpu: MRR@10:0.3535 R@1k: 97.97%
4_gpu: MRR@10:0.3539 R@1k: 97.99%

However a strange thing is that the mono gpu is a lot sparser than the others (2 times smaller than 2_gpu and almost 4 time smaller than 4_gpu). Probably needs better hyperparams for more gpu

**Comment by MXueguang** [comment]
Ah I guess the reason is I forgot to add loss scaling for flop loss:

https://github.com/texttron/tevatron/blob/fd396ffadd967febf64a15830e67aa913a1670e5/src/tevatron/modeling/encoder.py#L100-L101

**Comment by cadurosar** [comment]
Ah it could be! Another thing is that it would be good to be able to track document size/query size (during training and encoding), but I have no idea how to do it during training in the current framework, I can take a look to add it to encoding.

**Comment by MXueguang** [comment]
probably in `SpladeTrainer`. logging the size while computing the flops loss, based on `self.state.global_step`
